src/features/app-interfaces/data/index.ts

```diff
@@ -0,0 +1,95 @@
+import { ApplicationId } from '@/features/applications/data/types'
+import { AppInterfaceEntity, dbConnection } from '@/features/common/data/indexed-db'
+import { invariant } from '@/utils/invariant'
+import { createTimestamp, writableAtomCache } from '@/features/common/data'
+import { atom, Getter, Setter } from 'jotai'
+import { useAtomCallback } from 'jotai/utils'
+import { useCallback } from 'react'
+import { Arc32AppSpec } from '@/features/app-interfaces/data/types'
+
+const getAppInterface = async (applicationId: ApplicationId) => {
+  invariant(dbConnection, 'dbConnection is not initialised')
+  return await (await dbConnection).get('app-interfaces', applicationId)
+}
+const writeAppInterface = async (appInterface: AppInterfaceEntity) => {
+  invariant(dbConnection, 'dbConnection is not initialised')
+  await (await dbConnection).put('app-interfaces', appInterface)
+}
+const getAppInterfaces = async () => {
+  invariant(dbConnection, 'dbConnection is not initialised')
+  return await (await dbConnection).getAll('app-interfaces')
+}
+
+const createWritableAppInterfaceEntityAtom = (_: Getter, __: Setter, applicationId: ApplicationId) => {
+  const appInterfaceAtom = atom<AppInterfaceEntity | undefined | Promise<AppInterfaceEntity | undefined>>(getAppInterface(applicationId))
+
+  return atom(
+    (get) => get(appInterfaceAtom),
+    async (_, set, appInterface: AppInterfaceEntity) => {
+      await writeAppInterface(appInterface)
+      set(appInterfaceAtom, appInterface)
+    }
+  )
+}
+
+export const [appInterfacesAtom, getAppInterfaceAtom] = writableAtomCache(
+  createWritableAppInterfaceEntityAtom,
+  (applicationId: ApplicationId) => applicationId
+)
+
+export const useCreateAppInterface = () => {
```

Comment:
@neilcampbell I refactored this method to only for creating an app interface because I think that we won't need the logic for updating for a while. This will keep the code a bit simpler. I'll add the update logic as a separated method later on.